src/main/java/frc/robot/Drive.java

```diff
@@ -33,9 +35,10 @@ public void initialize() {
     public void execute() {
         double fwd = MathUtil.applyDeadband(fwdSupplier.getAsDouble()*Constants.MAX_VELOCITY,0.3);
         double str = MathUtil.applyDeadband(strSupplier.getAsDouble()*Constants.MAX_VELOCITY, 0.3);
-        double rcw = MathUtil.applyDeadband(rtSupplier.getAsDouble()*Constants.MAX_VELOCITY, 0.3);
-
-        ChassisSpeeds driveStates = drivetrain.createChassisSpeeds(fwd*(shouldFlip?1:-1), str*(shouldFlip?1:-1), -rcw, Constants.FIELD_RELATIVE);
+        double rcw;
+        if (autoAlign.get() == null) rcw = MathUtil.applyDeadband(rtSupplier.getAsDouble() * Constants.MAX_VELOCITY, 0.3);
```

Comment:
Get the value from the provider once at the top, to reduce the risk of it changing between readings.

Reply:
good call, done